### PR TITLE
Add code example to ArrayMesh class docs

### PR DESCRIPTION
**Pull request by lupoDharkael**
After my contribution with a micro-tutorial about how to use ArrayMesh [here](https://www.reddit.com/r/godot/comments/8pcypr/arraymesh_code_example_gdscript/), I decided to contribute it to the official docs as someone commented there.

### CODE REVIEW

File: doc/classes/ArrayMesh.xml [hunk 1]
```diff
@@ -3,6 +3,23 @@
...
 	<description>
+		The [code]ArrayMesh[/code] is used to construct a [Mesh] by specifying the attributes as arrays. The most basic example is the creation of a single triangle
+		[codeblock]
+		var vertices = PoolVector3Array()
+		vertices.push_back(Vector3(0,1,0))
+		vertices.push_back(Vector3(1,0,0))
+		vertices.push_back(Vector3(0,0,1))
+		# Initialize the ArrayMesh.
+		var arr_mesh = ArrayMesh.new()
+		var arrays = Array()
```
Comment (LikeLakers2) [inline doc/classes/ArrayMesh.xml]: Not sure if you need to use `Array()` here. `[]` in place of `Array()` probably works just fine.

Comment (lupoDharkael) [reply doc/classes/ArrayMesh.xml]: var arrays = Array() works too but I agree using [] is better. This code happens to be a conversion from a GDN C++ module I wrote and I did a semi-literal conversion.

**Review by mhilbrunner — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.0.7.

**Pull request merged**